cmd/teleporter-cli/transaction.go

```diff
@@ -30,46 +33,105 @@ var transactionCmd = &cobra.Command{
-				logger.Debug("Processing Warp log", zap.Any("log", log))
-
-				unsignedMsg, err := warp.UnpackSendWarpEventDataToMessage(log.Data)
-				cobra.CheckErr(err)
-
-				warpPayload, err := warpPayload.ParseAddressedCall(unsignedMsg.Payload)
-				cobra.CheckErr(err)
-
-				teleporterMessage, err := teleportermessenger.UnpackTeleporterMessage(warpPayload.Payload)
-				cobra.CheckErr(err)
-				logger.Info("Parsed Teleporter message",
-					zap.String("warpMessageID", unsignedMsg.ID().Hex()),
-					zap.String("teleporterMessageNonce", teleporterMessage.MessageNonce.String()),
-					zap.Any("message", teleporterMessage))
-			}
+		txHash := common.HexToHash(args[0])
+		if debug {
+			printTransaction(cmd, txHash)
+			traceTransaction(cmd, txHash)
 		}
+		checkReceipt(cmd, txHash)
 		cmd.Println("Transaction command ran successfully")
 	},
 }
 
+func checkReceipt(cmd *cobra.Command, txHash common.Hash) {
+	receipt, err := client.TransactionReceipt(context.Background(), txHash)
+	cobra.CheckErr(err)
+
+	for _, log := range receipt.Logs {
+		printTeleporterLogs(cmd, log)
+		printWarpLogs(cmd, log)
+	}
+}
+
+func printTeleporterLogs(cmd *cobra.Command, log *types.Log) {
+	if log.Address != teleporterAddress {
+		return
+	}
+
+	logJson, err := json.MarshalIndent(log, "", "  ")
+	cobra.CheckErr(err)
+
+	cmd.Println("Teleporter Log:\n" + string(logJson) + "\n")
+
+	event, err := teleporterABI.EventByID(log.Topics[0])
+	cobra.CheckErr(err)
+
+	out, err := teleportermessenger.FilterTeleporterEvents(log.Topics, log.Data, event.Name)
+	cobra.CheckErr(err)
+
+	outJson, err := json.MarshalIndent(out, "", "  ")
+	cobra.CheckErr(err)
+
+	cmd.Println(event.Name + " Log:")
+	cmd.Println(string(outJson) + "\n")
+}
+
+func printWarpLogs(cmd *cobra.Command, log *types.Log) {
+	if log.Address != common.HexToAddress(warpPrecompileAddress) {
+		return
+	}
+
+	logJson, err := json.MarshalIndent(log, "", "  ")
+	cobra.CheckErr(err)
+
+	cmd.Println("Warp Log:\n" + string(logJson) + "\n")
+
+	unsignedMsg, err := warp.UnpackSendWarpEventDataToMessage(log.Data)
+	cobra.CheckErr(err)
+	cmd.Println("Warp Message ID: " + unsignedMsg.ID().Hex())
+
+	warpPayload, err := warpPayload.ParseAddressedCall(unsignedMsg.Payload)
+	cobra.CheckErr(err)
+
+	warpPayloadJson, err := json.MarshalIndent(warpPayload, "", "  ")
+	cobra.CheckErr(err)
+	cmd.Println("Warp Payload:")
+	cmd.Println(string(warpPayloadJson))
+
+	teleporterMessage, err := teleportermessenger.UnpackTeleporterMessage(warpPayload.Payload)
+	cobra.CheckErr(err)
+
+	teleporterMessageJson, err := json.MarshalIndent(teleporterMessage, "", "  ")
+	cobra.CheckErr(err)
+
+	cmd.Println("Teleporter Message:")
+	cmd.Println(string(teleporterMessageJson))
+}
+
+func traceTransaction(cmd *cobra.Command, txHash common.Hash) {
+	var result interface{}
+	ct := "callTracer"
+	err := client.Client().Call(&result, "debug_traceTransaction", txHash.String(), tracers.TraceConfig{Tracer: &ct})
+	cobra.CheckErr(err)
```

Comment:
Rather than exiting with an error code here, it might be better to log that an error occurred but not exit. That way if the user does not have access to a debug-enabled API node, they can still use `printTransaction` when calling the CLI with the `--debug` option.